### ISSUE TITLE
Incorrectly flagging no_log_password on tasks that use vaulted passwords

### ISSUE THREAD

**Issue by RayMcAdmin** [ansible-community/ansible-lint#1595]
##### Summary

When using Red Hat Satellite Content Collection, many tasks require a password. This password is stored in the project in a vaulted file. Playbook correctly returns "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
Ansible lint comes back with "no-log-password password should not be logged." which, as far as I can tell, it is not being.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```ansible --version
ansible 2.10.10
  config file = /home/ray/Desktop/Gitlab stuff/Git Repos/Satellite-mk2/satellite/ansible.cfg
  configured module search path = ['/home/ray/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ray/.local/lib/python3.9/site-packages/ansible
  executable location = /home/ray/.local/bin/ansible
  python version = 3.9.5 (default, May 14 2021, 00:00:00) [GCC 10.3.1 20210422 (Red Hat 10.3.1-1)]
ansible-lint --version
ansible-lint 5.0.11 using ansible 2.10.10

```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```
- name: "Create Composite Content Views"
  redhat.satellite.content_view:
    username: "{{ satellite_deployment_admin_username }}"
    password: "{{ satellite_deployment_admin_password }}"
    server_url: "https://{{ satellite_deployment_hostname_full }}"
    name: "{{ item.name }}"
    organization: "{{ item.organization | default('ITS') }}"
    state: "{{ item.state | default('present') }}"
    composite: True
    components: "{{ item.components }}"
  loop: "{{ cvc }}"
  delegate_to: localhost

Password is stored in a vault encrypted file.


```

##### Desired Behaviour

Linting passes

##### Actual Behaviour

Linting fails with
```
roles/satellite-installer/tasks/configure_content_views.yaml:3: no-log-password password should not be logged.
roles/satellite-installer/tasks/configure_content_views.yaml:19: no-log-password password should not be logged.
```

**Comment by felixfontein** [comment]
Related: #1589
